core/src/pixelcolor/transparent_color.rs

```diff
@@ -0,0 +1,155 @@
+///         let inner = value.into_inner();
+///         let alpha = ((inner & 0xFF00) >> 8) as u8;
+///         TransparentGray { alpha, gray: Gray8::new((inner & 0xFF) as u8)}
+///     }
+/// }
+///
+/// // From<TransparentGray> is needed for PixelColor
+/// impl From<TransparentGray> for RawU16 {
+///     fn from(value: TransparentGray) -> Self {
+///         let TransparentGray { alpha, gray } = value;
+///         RawU16::new((alpha as u16) << 8 + gray.luma())
+///     }
+/// }
+///
+/// // Implement required methods
+/// impl TransparentColor<Gray8> for TransparentGray {
+///     fn blend_over(self, other: Gray8) -> Gray8 {
+///         let luma = (self.alpha as u16 * self.gray.luma() as u16) +
+///                    (255 - self.alpha as u16) * other.luma() as u16;
+///         Gray8::new((luma / 255) as u8)
+///     }
+/// }
+///
+/// // Make conversion available for `DrawTargetExt::color_converted`
+/// impl From<Gray8> for TransparentGray {
+///     fn from(value: Gray8) -> Self {
+///         TransparentGray { alpha: 255, gray: value }
+///     }
+/// }
+///
+/// // Mock transparent handling display
+/// pub struct Display {}
+/// impl DrawTarget for Display {
+///     type Color = TransparentGray;
+///     type Error = core::convert::Infallible;
+///
+///     fn draw_iter<I>(&mut self, pixels: I) -> Result<(), core::convert::Infallible>
+///     where
+///         I: IntoIterator<Item = Pixel<Self::Color>>,
+///     {
+///         for Pixel(point, color) in pixels.into_iter() {
+///             // mock existing pixel as middle gray
+///             let current_color = Gray8::new(127);
+///             // blend the color with existing color
+///             let final_color = color.blend_over(current_color);
+///             // ... draw final color on display
+///         }
+///         Ok(())
+///     }
+/// }
+/// impl OriginDimensions for Display {
+///     fn size(&self) -> Size {
+///         Size::new(300, 300)
+///     }
+/// }
+///
+/// // The image data.
+/// const DATA: &[u8] = &[
+///     0x80, 0xFF, 0xFF, 0x80, //
+///     0xFF, 0x00, 0x00, 0xFF, //
+///     0xFF, 0x00, 0x00, 0xFF, //
+///     0x80, 0xFF, 0xFF, 0x80, //
+/// ];
+///
+/// // Create a 4x4 non-transparent image
+/// let raw_image = ImageRaw::<Gray8>::new(DATA, Size::new(4,4)).unwrap();
+/// let image = Image::new(&raw_image, Point::zero());
+///
+/// // Create a transparent handling display
+/// let mut display = Display{};
+///
+/// // Draw a non-transparent object on a transparent handling display
+/// // Conversion is automatically handled by the `From` trait above and the `DrawTargetExt` trait
+/// image.draw(&mut display.color_converted()).unwrap();
+/// ```
+
+/// Transparent color trait.
+///
+/// This trait is used to represent color that handle transparency level.
+/// They can blend with other colors, transparent or nor.
+///
+/// Transparent colors are a special kind of [`PixelColor`].
+pub trait TransparentColor<C: PixelColor=Self>: PixelColor {
```

Comment:
We should find another name for this trait. To make it easier to implement code that works with transparent and opaque colors this trait can also be implemented for opaque colors:
```rust
impl TransparentColor for Rgb888 {
    fn blend_over(self, other: Rgb888) -> Rgb888 {
        self
    }
}
```
But the current name is a bit confusing, because `Rgb888` isn't transparent.

In other languages I would probably call this trait `BlendableColor`, but I think that using `able` is discouraged in some Rust API naming guideline.

Reply:
I don't know of any such recomendation